src/cpp/cpp_typecheck_code.cpp

```diff
@@ -191,8 +198,11 @@ void cpp_typecheckt::typecheck_switch(codet &code)
     assert(decl.operands().size()==1);
 
     // replace declaration by its symbol
-    assert(decl.op0().op0().id()==ID_symbol);
-    value = decl.op0().op0();
+    DATA_INVARIANT(
+      to_code_decl(to_code(to_unary_expr(decl).op())).symbol().id() ==
+        ID_symbol,
+      "declaration must have symbol as operand");
```

Comment:
This DATA_INVARIANT cannot fail, because `to_code_decl` will already fail its `code_declt::check(code);`

Reply:
removed